src/protocols/coda_transition_frontier.ml

```diff
@@ -2,6 +2,42 @@ open Core_kernel
 open Async_kernel
 open Pipe_lib.Strict_pipe
 
+module Transition_frontier_diff = struct
+  type 'a new_best_tip = {old_best_tip: 'a; new_best_tip: 'a}
+
+  type 'a new_root =
+    { old_root: 'a
+    ; new_root: 'a
+    ; added: 'a
+    ; garbage: 'a list
+    ; best_tip: 'a new_best_tip option }
+
+  type 'a t =
+    (* Added a node to the existing best tip without creating a new root *)
+    | Extend_best_tip of 'a
+    (* Added a node to the a new best tip without creating a new root *)
+    | New_best_tip of 'a new_best_tip
+    (* might not have been added to the best tip *)
+    | New_root of 'a new_root
```

Comment:
may as well use anonymous records for these two since I don't think we'd ever use them separately but if you didn't:

the style nit is prefer

```
module New_best_tip = struct
   type 'a t = (* ... *)
end
```

to named modules

Also, I'd stick `@@deriving sexp` on the diff because we'll probably want that